packages/amplify-graphql-schema-generator/src/schema-generator/generate-schema.ts

```diff
@@ -209,6 +218,12 @@ const constructEnumType = (type: EnumType): EnumWrapper => {
   return enumType;
 };
 
+const validateEnumValues = (values: string[]): boolean => {
+  const regex = new RegExp(/^[_A-Za-z][_0-9A-Za-z]*$/);
```

Comment:
I believe the spec requires there to be at least one `NameContinue` after `NameStart`, so the regex quantifier should be `+` rather than `*`:

```suggestion
  const regex = new RegExp(/^[_A-Za-z][_0-9A-Za-z]+$/);
```

We should add a test for single-character enum values to validate.

Reply:
I don't see an issue with the regex expression. There are two character classes in this regex expression. [_A-Za-z] verifies the first character (NameStart) and [_0-9A-Za-z]* confirms the rest of the string value (NameContinue). Added a test with single character enum value.

I have confirmed with the GraphQL 2018 spec (that's what AppSync supports) as well.
https://spec.graphql.org/June2018/#Name